shared/src/shared.rs

```diff
@@ -200,6 +205,10 @@ impl Shared {
     pub fn store(&self) -> &ChainDB {
         &self.store
     }
+
+    pub fn header_map_config(&self) -> &HeaderMapConfig {
```

Comment:
I don't think we need a fn to get `header_map_config` here, the config should be passed to `Shared::init` directly like other config objects